packages/core-ethereum/src/index.ts

```diff
@@ -373,6 +396,19 @@ export default class HoprCoreEthereum extends EventEmitter {
         error: err
       }
     }
+
+    // Give other tasks CPU time to happen
+    // Push update of database to end of next event loop iteration
+    await setImmediate()
```

Comment:
I don't think we need to be that granular in defering computation. If the block of computation is long then that's how it is. We just must ensure that between iterations we unblock. Even worse, by going into that fine granularity we could invoke inconsistencies in our view of the data.

Reply:
At this point, it is indeed not beneficial